### ISSUE TITLE
Does latest version 1.120.1 support ConfigConnector namespaced mode?

### ISSUE THREAD

**Issue by kallentrustle** [GoogleCloudPlatform/k8s-config-connector#2362]
### Describe your question

Today I installed version 1.120.1 manually on my GKE cluster. When I applied my ConfigConnector manifest, `mode: namespaced`, I got this error:

```
The ConfigConnector "configconnector.core.cnrm.cloud.google.com" is invalid:
* <nil>: Invalid value: "": "spec" must validate at least one schema (anyOf)
* spec.mode: Unsupported value: "namespaced": supported values: "cluster"

apiVersion: core.cnrm.cloud.google.com/v1beta1
kind: ConfigConnector
metadata:
  name: configconnector.core.cnrm.cloud.google.com
spec:
  mode: namespaced
  googleServiceAccount: "sa-mygccname@PROJECT_ID.iam.gserviceaccount.com"
  stateIntoSpec: Absent
 ```
I looked at the release notes, and didn't see anything about namespaced mode no longer being supported. GCP docs say you can use namespaced: https://cloud.google.com/config-connector/docs/how-to/install-namespaced

On a different GKE cluster where ConfigConnector is installed as an addon, version 1.102.0, I've been successfully using (testing) it in namespaced mode. I'm hoping to use this same setup in the newest version ...

**Comment by justinsb** [comment]
Thanks for reporting.  Namespace mode is certainly supported - it is in fact generally the mode we recommend over cluster mode.  I'll take a look to see what could be going on!

**Comment by justinsb** [comment]
Can you possibly check the `name: configconnectors.core.cnrm.cloud.google.com` CustomResourceDefinition?

This validation error looks to be schema validation, but the schema supports `cluster` and `namespaced` and git blame shows that it hasn't changed for years (I changed the description format a few months ago, but that _shouldn't_ change the schema): 

https://github.com/GoogleCloudPlatform/k8s-config-connector/blame/378a294517d6ceb2ac794ed690d2f4530e9f20ba/operator/config/crd/bases/core.cnrm.cloud.google.com_configconnectors.yaml#L82-L85


My guess is that either the CRD you are applying is somehow corrupted on disk (which seems unlikely), or that the CRD on the cluster is somehow different (which I haven't seen, but seems the most likely).  If you `kubectl get customresourcedefinition configconnectors.core.cnrm.cloud.google.com -oyaml` and attach/paste the output maybe that will help us figure it out.

**Comment by kallentrustle** [comment]
I see what I did incorrectly. Following https://cloud.google.com/config-connector/docs/how-to/install-namespaced, I skipped over the first apply of creating the `ConfigConnector` as stated. I sort of jumped ahead to create a manifest for `ConfigConnector` with a `spec.googleServiceAccount` in it. But `spec.googleServiceAccount` belongs in a `ConfigConnectorContext` manifest, not `ConfigConnector`.

The manifest I posted above would have worked if I had omitted `spec.googleServiceAccount`. Tonight after successfully creating the `ConfigConnector`, I then created my first `ConfigConnectorContext` that includes my staging namespace and the `spec.googleServiceAccount` bound to it.

tl;dr, pebcak! The error message threw me for a loop. Thank you for responding so quickly!
